apps/src/code-studio/header.js

```diff
@@ -146,6 +157,12 @@ header.build = function (scriptData, stageData, progressData, currentLevelId, pu
   }
 };
 
+const ConnectedShareDialog = connect(state => ({
+  isOpen: state.header.shareDialog.isOpen,
+}), dispatch => ({
+  onClose: () => dispatch(hideShareDialog())
+}))(ShareDialog);
```

Comment:
This should be easy for ShareDialog, but I see no obvious way to do this with a single ConnectedPublishDialog component because we are connecting two different parts of the redux state tree to it. I'd love to extract and share the connect logic if you know how. One idea is to provide a different sub-tree as the store passed to the `<Provider>` tag, but i'm not sure if that's possible.